### PR TITLE
Debugger: Add plan info to debug info

### PR DESCRIPTION
**Pull request by kraftbj**
To aid HEs in debugging issues.

### CODE REVIEW

File: class.jetpack-debugger.php [hunk 1]
```diff
@@ -18,6 +18,31 @@ private static function is_jetpack_support_open() {
...
+		switch ( $plan ){
+			case 'jetpack_free' :
+			default :
+				$plan = 'Free';
+				break;
+			case 'jetpack_premium' :
+			case 'jetpack_premium_monthly' :
+				$plan = 'Premium';
+				break;
+			case 'jetpack_personal' :
```
Comment (georgestephanis) [inline class.jetpack-debugger.php]: Out of curiosity, is there a reason that we're ordering Personal after Premium?  Wouldn't it make more sense to order from lowest to highest?

Also, would it be at all useful for happiness to be able to easily tell if it's monthly/annual, and the expiration date?  Or do y'all just care what it is now?

**Comment by samhotchkiss** [comment]
I super simplified this... I don't see any reason for us to modify the plan names-- why not just pass the plan name string?  This is more futureproof and would potentially be more helpful in figuring an issue out.

cc: @kraftbj @jeherve

**Comment by samhotchkiss** [comment]
If y'all are good with this, merge at will.

File: class.jetpack-debugger.php (hunk 1 above)
Comment (kraftbj) [inline class.jetpack-debugger.php]: Just the plan that the site thinks it has is enough for us. I had Personal first for performance (started this with if statements and Personal, I'd expect, to be the most purchased plan.

The KISS update negates the need for that though.

**Comment by kraftbj** [comment]
After further reflection, I'd like to change this a bit to standardize plan type by working with https://github.com/Automattic/jetpack/blob/997a48a312c6cef0a4cd97998dba1176957eba9a/class.jetpack.php#L1232 to include the detail we'd need here.

I'll tag for review tomorrow/Monday after tweaking it.

**Comment by zinigor** [comment]
@kraftbj I have added supported features to the output, is that what you had in mind?

**Comment by kraftbj** [comment]
@zinigor No, I'm going to put this on hold until I have a chance to make the changes to add to the discussion.

**Comment by kraftbj** [comment]
@zinigor I reworked it some. The thing I'm trying to balance out is simplifying what plan gets passed in (just the type of plan, annual vs monthly doesn't matter) to save the parser on our end from needing to be tweaked too much. What do you think?

File: class.jetpack-debugger.php [hunk 2]
```diff
@@ -18,6 +18,12 @@ private static function is_jetpack_support_open() {
 		}
 	}
 
+	private static function what_jetpack_plan() {
+		$plan = Jetpack::get_active_plan();
+		$plan = $plan['class'];
```
Comment (georgestephanis) [inline class.jetpack-debugger.php]: Should we add a fallback value if `$plan['class']` is empty?  Like if someone adds a new plan down the road and forgets to include a class?

Comment (dereksmart) [inline class.jetpack-debugger.php]: I agree.  The way `get_active_plan()` is written it makes it possible for this value to not be set, and in that case would throw a notice.  Let's add a fallback of `unknown` if not set to future-proof.

**Comment by dereksmart** [comment]
@kraftbj pushed 261a1145f2ca312a4956f28493609c22cb16570c to address george's comment, which I agree with. Feel free to merge if you're happy.

**Pull request merged**